src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -395,6 +382,64 @@ class npc_fel_guard_hound : public CreatureScript
     }
 };
 
+enum Beacon
+{
+    NPC_STONESCHYE_WHELP        = 16927,
+};
+
+class go_beacon : public GameObjectScript
+{
+public:
+    go_beacon() : GameObjectScript("go_beacon") { }
+
+    struct go_beaconAI : public GameObjectAI
+    {
+        go_beaconAI(GameObject* gameObject) : GameObjectAI(gameObject) { }
+
+        std::list<Creature*> creatureList;
+
+        void OnStateChanged(uint32 state, Unit*  /*unit*/) override
+        {
+            if (state == GO_ACTIVATED)
+            {
+                me->GetCreaturesWithEntryInRange(creatureList, 40, NPC_STONESCHYE_WHELP);
+                {
+                    uint8 i = 0;
+                    for (Creature* whelp : creatureList)
+                    {
+                        if (whelp->IsAlive() && !whelp->IsInCombat() && whelp->GetMotionMaster()->GetCurrentMovementGeneratorType() != HOME_MOTION_TYPE)
+                        {
+                            float o = i * 9; i++;
+                            whelp->GetMotionMaster()->MovePoint(0, me->GetPositionX() + urand(3, 4) * cos(o) , me->GetPositionY() + urand(3, 4) * sin(o), me->GetPositionZ());
```

Comment:
final position should be calculated with a straight line from npcPos to goPos, without random and angles involved. You could try with GetFirstCollisionPosition and move there.

Reply:
I was wondering how I can achieve that, will look into it. Thanks.